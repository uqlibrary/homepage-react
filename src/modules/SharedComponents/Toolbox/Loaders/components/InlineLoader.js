import React from 'react';
import PropTypes from 'prop-types';
import CircularProgress from '@mui/material/CircularProgress';
import Grid from '@mui/material/Grid';
import Hidden from '@mui/material/Hidden';
import Typography from '@mui/material/Typography';

export class InlineLoader extends React.Component {
    static propTypes = {
        message: PropTypes.string,
        classes: PropTypes.object,
    };

    static defaultProps = {
        message: 'Loading',
    };

    render() {
        return (
            <div style={{ padding: 8 }}>
                <Grid
                    container
                    direction={'row'}
                    spacing={1}
                    justifyContent="center"
                    alignItems="center"
                    alignContent={'center'}
                >
                    <Hidden smUp>
                        <Grid item xs />
                    </Hidden>
                    <Grid item xs={'auto'} style={{ textAlign: 'center' }}>
                        <CircularProgress
                            sx={{ color: 'primary.light' }}
                            className={this.props.classes.circular}
                            size={18}
                            thickness={2}
                            aria-labelledby="loading-icon"
                        />
                    </Grid>
                    <Grid item xs={'auto'} style={{ textAlign: 'center' }}>
                        <Typography
                            id="loading-icon"
                            sx={{ color: 'primary.light', fontSize: '1.33rem' }}
                            variant={'h5'}
                            component={'span'}
                        >
                            {this.props.message}
                        </Typography>
                    </Grid>
                    <Hidden smUp>
                        <Grid item xs />
                    </Hidden>
                </Grid>
            </div>
        );
    }
}

export default InlineLoader;
