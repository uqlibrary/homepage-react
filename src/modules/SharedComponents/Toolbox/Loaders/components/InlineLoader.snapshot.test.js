import { InlineLoader } from './InlineLoader';

function setup(testProps = {}) {
    const props = {
        ...testProps,
        classes: {
            text: {
                fontWeight: 200,
                margin: '24px 0',
            },
        },
    };
    return getElement(InlineLoader, props);
}

describe('Component InlineLoader', () => {
    it('should render as expected', () => {
        const props = {
            message: 'This is a tst',
        };
        const wrapper = setup({ ...props });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
