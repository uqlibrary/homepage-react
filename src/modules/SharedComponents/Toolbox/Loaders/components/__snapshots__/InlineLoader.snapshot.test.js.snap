// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component InlineLoader should render as expected 1`] = `
<div
  style={
    Object {
      "padding": 8,
    }
  }
>
  <WithStyles(ForwardRef(Grid))
    alignContent="center"
    alignItems="center"
    container={true}
    direction="row"
    justify="center"
    spacing={1}
  >
    <Hidden
      smUp={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      />
    </Hidden>
    <WithStyles(ForwardRef(Grid))
      item={true}
      style={
        Object {
          "textAlign": "center",
        }
      }
      xs="auto"
    >
      <WithStyles(ForwardRef(CircularProgress))
        color="primary"
        size={18}
        thickness={2}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      style={
        Object {
          "textAlign": "center",
        }
      }
      xs="auto"
    >
      <WithStyles(ForwardRef(Typography))
        color="primary"
        component="span"
        style={
          Object {
            "fontSize": "1.33rem",
          }
        }
        variant="h5"
      >
        This is a tst
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <Hidden
      smUp={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      />
    </Hidden>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
